### PR TITLE
feat(expr): add `array_sort`

### LINKED ISSUE #11996: Implement array_{min/max/sum/sort}

### PR DESCRIPTION
**Pull request by xzhseh**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

add `array_sort` & related #11996

## Checklist

- [ ] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: e2e_test/batch/functions/array_sort.slt.part [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+query I
+select array_sort(array['114514🤔️1919810', '113514🥵1919810', '112514😅1919810']);
+----
+{112514😅1919810,113514🥵1919810,114514🤔️1919810}
+
+query I
+select array_sort(array[3, 2, NULL, 1, NULL]);
+----
+{1,2,3}
```
Comment (wangrunji0408) [inline e2e_test/batch/functions/array_sort.slt.part:34]: Should we keep the nulls?

Comment (xzhseh) [reply e2e_test/batch/functions/array_sort.slt.part:34]: But if choosing to keep the nulls, what should be the correct precedence? (i.e., in the front,  at the end)
Also I'm wondering is it helpful to the user if the nulls are kept🤔️.

Comment (TennyZhuang) [reply e2e_test/batch/functions/array_sort.slt.part:34]: > `NULLS FIRST is the default for DESC order`

Comment (wangrunji0408) [reply e2e_test/batch/functions/array_sort.slt.part:34]: > Also I'm wondering is it helpful to the user if the nulls are kept🤔️.

It's hard to say. But users will definitely be surprised if some elements are gone after sort. 😇

**Comment by xzhseh** [comment]
The updated version changes `DefaultOrdered(ScalarRefImpl<'_>)` to `DefaultOrdered(DatumRef<'_>)`, thus keep NULLs at the very end after sorting by default 🥵.

**Review by wangrunji0408 — APPROVED** [review]
LGTM!

**Pull request merged**